RecoVertex/Configuration/python/RecoVertex_cff.py

```diff
@@ -33,36 +33,47 @@
                           )
 vertexreco = cms.Sequence(vertexrecoTask)
 
-#timing
-from RecoVertex.PrimaryVertexProducer.TkClusParameters_cff import DA2D_vectParameters
-DA2D_vectParameters.TkDAClusParameters.verbose = cms.untracked.bool(False)
-unsortedOfflinePrimaryVertices4D = unsortedOfflinePrimaryVertices.clone( verbose = cms.untracked.bool(False),
-                                                                         TkClusParameters = DA2D_vectParameters )
-unsortedOfflinePrimaryVertices4D.TkFilterParameters.minPt = cms.double(0.0)
-unsortedOfflinePrimaryVertices4D.TrackTimesLabel = cms.InputTag("trackTimeValueMapProducer:generalTracksConfigurableFlatResolutionModel")
-unsortedOfflinePrimaryVertices4D.TrackTimeResosLabel = cms.InputTag("trackTimeValueMapProducer:generalTracksConfigurableFlatResolutionModelResolution")
-offlinePrimaryVertices4D=sortedPrimaryVertices.clone(vertices="unsortedOfflinePrimaryVertices4D", particles="trackRefsForJetsBeforeSorting4D", trackTimeTag=cms.InputTag("trackTimeValueMapProducer","generalTracksConfigurableFlatResolutionModel"),trackTimeResoTag=cms.InputTag("trackTimeValueMapProducer","generalTracksConfigurableFlatResolutionModelResolution"),assignment=dict(useTiming=True))
-offlinePrimaryVertices4DWithBS=sortedPrimaryVertices.clone(vertices="unsortedOfflinePrimaryVertices4D:WithBS", particles="trackRefsForJetsBeforeSorting4D", trackTimeTag=cms.InputTag("trackTimeValueMapProducer","generalTracksConfigurableFlatResolutionModel"),trackTimeResoTag=cms.InputTag("trackTimeValueMapProducer","generalTracksConfigurableFlatResolutionModelResolution"),assignment=dict(useTiming=True))
-
-trackWithVertexRefSelectorBeforeSorting4D = trackWithVertexRefSelector.clone(vertexTag="unsortedOfflinePrimaryVertices4D")
-trackWithVertexRefSelectorBeforeSorting4D.ptMax=9e99
-trackWithVertexRefSelectorBeforeSorting4D.ptErrorCut=9e99
-trackRefsForJetsBeforeSorting4D = trackRefsForJets.clone(src="trackWithVertexRefSelectorBeforeSorting4D")
-
-from SimTracker.TrackerHitAssociation.tpClusterProducer_cfi import tpClusterProducer
-from SimTracker.TrackAssociatorProducers.quickTrackAssociatorByHits_cfi import quickTrackAssociatorByHits
-from SimTracker.TrackAssociation.trackTimeValueMapProducer_cfi import trackTimeValueMapProducer
+#modifications for timing
+from RecoVertex.Configuration.RecoVertex_phase2_timing_cff import *
 _phase2_tktiming_vertexrecoTask = cms.Task( vertexrecoTask.copy() ,
                                             tpClusterProducer ,
                                             quickTrackAssociatorByHits ,
```

Comment:
remind me please why these truth matching modules are sitting in the RECO piece?

Reply:
This is part of the "fast simulation" of the MTD used for the technical proposal, and which we want to keep around just a little while longer to facilitate direct comparisons.  After the TDR release it can be permanently removed.